StyleCop.Analyzers/StyleCop.Analyzers.CodeFixes/ReadabilityRules/SA1129CodeFixProvider.cs

```diff
@@ -58,12 +58,32 @@ public override Task RegisterCodeFixesAsync(CodeFixContext context)
         private static SyntaxNode GetReplacementNode(SyntaxNode node)
         {
             var newExpression = (ObjectCreationExpressionSyntax)node;
+            var nameSyntax = newExpression.Type as NameSyntax;
+            var identifierName = NameSyntaxHelpers.ToNormalizedString(nameSyntax);
 
-            return SyntaxFactory.DefaultExpression(newExpression.Type)
+            SyntaxNode replacement = null;
+            if (identifierName.EndsWith(nameof(CancellationToken), System.StringComparison.OrdinalIgnoreCase))
```

Comment:
:exclamation: I overlooked this part originally. This check needs to check the *symbol*, not the syntax.